### PR TITLE
Update to use go1.10.1

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
Update to use go1.10.1

**Release note**:
```release-note
Update to use go1.10.1
```

### CODE REVIEW

**Comment by ixdy** [comment]
/hold

basically LGTM. I'm going to wait and see if the tests explode, and if not, I will build the crossbuild image.

**Comment by ixdy** [comment]
it looks like https://github.com/kubernetes/kubernetes/pull/60373 might be a blocker?

**Comment by cblecker** [comment]
I don't think so.. I checked out that PR locally, and it didn't fix this particular failure.
Created https://github.com/kubernetes/kubernetes/issues/60598 for it.

**Comment by cblecker** [comment]
Okay, so we can probably hold on building the images yet. Looks like we may need to wait for an upstream golang fix based on @liggitt's diagnosis here: https://github.com/kubernetes/kubernetes/issues/60598#issuecomment-369476335

**Comment by BenTheElder** [comment]
/cc
we'll want to upgrade some test-infra images too.

File: build/root/WORKSPACE [hunk 1]
```diff
@@ -1,8 +1,8 @@
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "66282d078c1847c2d876c02c5dabd4fd57cc75eb41a9668a2374352fa73b4587",
-    strip_prefix = "rules_go-ff7e3364d9383cf14155f8c2efc87218d07eb03b",
-    urls = ["https://github.com/bazelbuild/rules_go/archive/ff7e3364d9383cf14155f8c2efc87218d07eb03b.tar.gz"],
+    sha256 = "1cac93a87939915f26844b92df4f6b946f68707a4830a489c3382223e7913fbd",
+    strip_prefix = "rules_go-78c989a6002e1ee28f3a18e3d8d2787feb0fe3a6",
+    urls = ["https://github.com/bazelbuild/rules_go/archive/78c989a6002e1ee28f3a18e3d8d2787feb0fe3a6.tar.gz"],
```
Comment (ixdy) [inline build/root/WORKSPACE]: maybe we should just switch to the release tarball directly? i.e. https://github.com/bazelbuild/rules_go/releases/download/0.10.2/rules_go-0.10.2.tar.gz.

this won't need the strip_prefix, either.
```
$ sha256sum rules_go-0.10.2.tar.gz 
4b2c61795ac2eefcb28f3eb8e1cb2d8fb3c2eafa0f6712473bc5f93728f38758  rules_go-0.10.2.tar.gz
```

File: build/root/WORKSPACE [hunk 2]
```diff
@@ -38,15 +38,15 @@ http_archive(
 
 load("@bazel_skylib//:lib.bzl", "versions")
 
-versions.check(minimum_bazel_version = "0.10.0")
+versions.check(minimum_bazel_version = "0.10.1")
```
Comment (ixdy) [inline build/root/WORKSPACE]: is there a particular need for `0.10.1`?

Comment (cblecker) [reply build/root/WORKSPACE]: Oops. Just a case of version number blur when updating stuff.

**Comment by cblecker** [comment]
@ixdy Bazel test is green 🎉! We should be good to push images.

**Comment by ixdy** [comment]
```
$ docker pull golang:1.10.1
Error response from daemon: manifest for golang:1.10.1 not found
```
???

**Comment by ixdy** [comment]
oh, that seems to match https://hub.docker.com/r/library/golang/tags/, but disagree with https://hub.docker.com/_/golang/.

**Comment by ixdy** [comment]
https://github.com/docker-library/golang/issues/216

**Comment by cblecker** [comment]
@BenTheElder the `k8s.gcr.io/kube-cross:v1.10.1-1` image needs to be pushed first before the cross/verify jobs will pass.

@ixdy Looks like `golang:1.10.1` is available now.

**Comment by ixdy** [comment]
image built and pushed.

/retest
/lgtm

**Comment by ixdy** [comment]
something weird is going on; the new tag appears under gcr.io/google-containers/kube-cross, but not k8s.gcr.io/kube-cross. I'm hoping it's an eventual-consistency thing that'll work itself out in a little bit.

**Pull request merged**